csrc/scheduler/resize.cpp

```diff
@@ -335,7 +346,51 @@ void ResizeScheduler::schedule(Fusion* fusion, const HeuristicParams* params) {
   const int64_t bdimx = 128;
 
   // Make sure the DID ID located at the outermost position
-  const auto outermost_pos = scheduler_utils::reorderDevicesToOuter(ref_tv);
+  auto outermost_pos = scheduler_utils::reorderDevicesToOuter(ref_tv);
+
+  // [DID, ..., ...]
+  //        ^
+  //        +--- outermost_pos
+
+  // Move the static repeat ID to the outermost position if
+  // detected. The repeat ID then just remains there with no
+  // scheduling.
+  bool repeat_id_moved_to_outermost = false;
+  if (static_repeat_info.has_value()) {
+    NVF_ERROR(ref_tv == static_repeat_info->repeat_output_tv);
+    auto ref_repeat_id_it = std::find_if(
+        ref_tv->getLoopDomain().begin(),
+        ref_tv->getLoopDomain().end(),
+        [&](IterDomain* loop_id) {
+          return id_model->idGraph(IdMappingMode::EXACT)
+              .disjointValSets()
+              .strictAreMapped(loop_id, static_repeat_info->reshape_repeat_id);
+        });
+    // Gives up if the repeat ID is not found. Unclear if this could
+    // actually happen, though.
```

Comment:
I'm fine with this as-is for now.
Should we try to assert on that so we would get a better idea about this?

Reply:
I also thought about that, but an assert would mean the user would see a failure, whereas just skipping this would still work functionally correctly. I think the latter would be a better user experience.